examples/keras_io/tensorflow/nlp/multi_label_classification.py

```diff
@@ -0,0 +1,426 @@
+sigmoid. Researchers have used other combinations of loss function and activation
+function as well. For example, in [Exploring the Limits of Weakly Supervised Pretraining](https://arxiv.org/abs/1805.00932),
+Mahajan et al. used the softmax activation function and cross-entropy loss to train
+their models.
+
+There are several options of metrics that can be used in multi-label classification.
+To keep this code example narrow we decided to use the
+[binary accuracy metric](https://keras.io/api/metrics/accuracy_metrics/#binaryaccuracy-class).
+To see the explanation why this metric is used we refer to this
+[pull-request](https://github.com/keras-team/keras-io/pull/1133#issuecomment-1322736860).
+There are also other suitable metrics for multi-label classification, like
+[F1 Score](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/F1Score) or
+[Hamming loss](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/HammingLoss).
+"""
+
+epochs = 20
+
+shallow_mlp_model = make_model()
+shallow_mlp_model.compile(
+    loss="binary_crossentropy", optimizer="adam", metrics=["binary_accuracy"]
+)
+
+history = shallow_mlp_model.fit(
+    train_dataset, validation_data=validation_dataset, epochs=epochs
+)
+
+
+def plot_result(item):
+    plt.plot(history.history[item], label=item)
+    plt.plot(history.history["val_" + item], label="val_" + item)
+    plt.xlabel("Epochs")
+    plt.ylabel(item)
+    plt.title("Train and Validation {} Over Epochs".format(item), fontsize=14)
+    plt.legend()
+    plt.grid()
+    plt.show()
+
+
+plot_result("loss")
+plot_result("binary_accuracy")
+
+"""
+While training, we notice an initial sharp fall in the loss followed by a gradual decay.
+"""
+
+"""
+### Evaluate the model
+"""
+
+_, binary_acc = shallow_mlp_model.evaluate(test_dataset)
+print(f"Categorical accuracy on the test set: {round(binary_acc * 100, 2)}%.")
+
+"""
+The trained model gives us an evaluation accuracy of ~99%.
+"""
+
+"""
+## Inference
+
+An important feature of the
+[preprocessing layers provided by Keras](https://keras.io/guides/preprocessing_layers/)
+is that they can be included inside a `keras.Model`. Note however that the [TextVectorization 
+layer](https://keras.io/keras_core/api/layers/preprocessing_layers/text/text_vectorization/)
+ uses TensorFlow internally. It cannot be used as part of a compiled computation graph 
+of a model with any backend other than TensorFlow. We will export an inference model
+by including the `text_vectorization` layer on top of `shallow_mlp_model`. This will
+allow our inference model to directly operate on raw strings.
+
+**Note** that during training it is always preferable to use these preprocessing
+layers as a part of the data input pipeline rather than the model to avoid
+surfacing bottlenecks for the hardware accelerators. This also allows for
+asynchronous data processing.
+"""
+
+# Create a model for inference.
```

Comment:
Since this won't work for other backends, better to show an example where we map the text_vectorizer into the tf.data Dataset. We need the code to run with all backends.